### PR TITLE
fix: Autocomplete does not suggest same function name that belongs to…

### PR DESCRIPTION
**Pull request by Druthi**
## Description
fix: Autocomplete does not suggest same function name that belongs to a different object

Fixes # (issue)
[#17059](https://app.zenhub.com/workspaces/fe-coders-pod-6167d1b64ee5f20014242ab7/issues/gh/appsmithorg/appsmith/17059)

image

## Type of change

> Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
- Cypress

### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

**Review by ApekshaBhosale — APPROVED** [review]

File: app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/Autocomplete/Autocomplete_JS_spec.ts [hunk 1]
```diff
@@ -276,4 +276,44 @@ describe("Autocomplete tests", () => {
       agHelper.GetNAssertElementText(CommonLocators._hints, "geolocation");
     });
   });
+
+  it.only("9. Bug #17059 Autocomplete does not suggest same function name that belongs to a different object", () => {
```
Comment (Aishwarya-U-R) [inline app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/Autocomplete/Autocomplete_JS_spec.ts:280]: Can we please remove '.only' here?

File: app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/Autocomplete/Autocomplete_JS_spec.ts [hunk 2]
```diff
@@ -276,4 +276,44 @@ describe("Autocomplete tests", () => {
...
+      toRun: false,
+      shouldCreateNewJSObj: true,
+      prettify: false,
+    });
+
+    // focus on 5th line
+    agHelper.GetNClick(jsEditor._lineinJsEditor(5));
+
+    // autocomplete verification
+    agHelper.TypeText(CommonLocators._codeMirrorTextArea, "JSObject1.");
```
Comment (Aishwarya-U-R) [inline app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/Autocomplete/Autocomplete_JS_spec.ts:303]: Here 'JSObject1.' is the object created in 1st case. Is this intended? If so, the JSObjects being created in this case are not used?

File: app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/Autocomplete/Autocomplete_JS_spec.ts [hunk 3]
```diff
@@ -276,4 +276,44 @@ describe("Autocomplete tests", () => {
...
+
+    agHelper.GetNAssertElementText(
+      CommonLocators._hints,
+      "myFun1.data",
+      "have.text",
+      0,
+    );
+
+    agHelper.GetNAssertElementText(
```
Comment (Aishwarya-U-R) [inline app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/Autocomplete/Autocomplete_JS_spec.ts:312]: Can we add similar assertions into another JSObject also - to be sure?

**Review by Aishwarya-U-R — COMMENTED** [review]
Cypress looks good

**Comment by ramsaptami** [comment]
Gtg to release
CC: @Druthi

**Review by rimildeyjsr — APPROVED** [review]

**Pull request merged**